src/Model/Profile.php

```diff
@@ -292,14 +304,16 @@ private static function sidebar(App $a, array $profile, $block = 0, $show_connec
 		$subscribe_feed_link = null;
 		$wallmessage_link = null;
 
+		// Who is the logged-in user to this profile?
 		$visitor_contact = [];
 		if (!empty($profile['uid']) && self::getMyURL()) {
 			$visitor_contact = Contact::selectFirst(['rel'], ['uid' => $profile['uid'], 'nurl' => Strings::normaliseLink(self::getMyURL())]);
 		}
 
+		// Who is this profile to the logged-in user?
 		$profile_contact = [];
-		if (!empty($profile['cid']) && self::getMyURL()) {
-			$profile_contact = Contact::selectFirst(['rel'], ['id' => $profile['cid']]);
+		if (!empty($contact) && self::getMyURL()) {
+			$profile_contact = $contact['rel'];
```

Comment:
The `$profile_contact` is an array that contains the contact data of the currently displayed contact. I would suggest changing the line 277 so that it fetches the `$profile_contact` there and to replace the newly created `$contact` variable with the existing variable `$profile_contact`.

And by thinking about the existing code, I think that there is a lot of crap in it :-) (Not your changes, I refer to the code you are replacing)